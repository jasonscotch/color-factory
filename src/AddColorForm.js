import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';

const AddColorForm = ({ addColor }) => {
  const [formData, setFormData] = useState({name: "", hex:"#ffffff"});
  const navigate = useNavigate();

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData(prevFormData => ({
        ...prevFormData,
        [name]: value
    }));
  }
  
  const handleSubmit = (e) => {
    e.preventDefault();
    addColor({[formData.name]: formData.hex});
    navigate('/colors');
  };

  const {name, hex} = formData;

  return (
    <div>
      <h2>Add A Color</h2>
      <form onSubmit={handleSubmit}>
        <label htmlFor='name'>Color Name:</label>
        <input
          id='name'
          placeholder='Enter a color name here'
          name='name'
          value={name}
          onChange={handleChange}
        />
        <label htmlFor='hex'>Color:</label>
        <input
          id='hex'
          type='color'
          name='hex'
          value={hex}
          onChange={handleChange}
        />
        <button type='submit'>Submit</button>
      </form>
    </div>
  );
}

export default AddColorForm;

