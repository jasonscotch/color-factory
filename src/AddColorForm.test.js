import { render, fireEvent } from '@testing-library/react';
import AddColorForm from './AddColorForm';
import { MemoryRouter } from 'react-router-dom';

it('renders AddColorForm correctly', () => {
  render(
    <MemoryRouter>
        <AddColorForm />
    </MemoryRouter>
    );
});

it("matches snapshot", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <AddColorForm />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it('adds a new color on form submission', () => {
    const mockAddColor = jest.fn();

    const {getByLabelText, queryByText } = render(
        <MemoryRouter>
            <AddColorForm addColor={mockAddColor} />
        </MemoryRouter>
    );

    const input1 = getByLabelText('Color Name:');
    const input2 = getByLabelText('Color:');
    const submitButton = queryByText('Submit');

    fireEvent.change(input1, { target: { value: 'red' } });
    fireEvent.change(input2, { target: { value: '#ff0000' } });
    fireEvent.click(submitButton);

    expect(mockAddColor).toHaveBeenCalledWith({
        red: '#ff0000'
    });

});
