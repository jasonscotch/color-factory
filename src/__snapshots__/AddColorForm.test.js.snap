// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Add A Color
    </h2>
    <form>
      <label
        for="name"
      >
        Color Name:
      </label>
      <input
        id="name"
        name="name"
        placeholder="Enter a color name here"
        value=""
      />
      <label
        for="hex"
      >
        Color:
      </label>
      <input
        id="hex"
        name="hex"
        type="color"
        value="#ffffff"
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
